shared/httputil/httputil.go

```diff
@@ -5,6 +5,30 @@ import (
 	"strings"
 )
 
+// parseForwardedHeader parses the RFC7239 Forwarded header from a http.Request
+// and return a slice of the values queried from the header
+func parseForwardedHeader(r *http.Request, token string) (val []string) {
+	for _, v := range r.Header["Forwarded"] {
+		options := strings.Split(v, ";")
+		for _, o := range options {
+			keyvalue := strings.Split(o, "=")
+			key, value := strings.TrimSpace(keyvalue[0]), strings.TrimSpace(keyvalue[1])
+			key = strings.ToLower(key)
+			if key == token {
+				val = append(val, value)
+			}
+		}
+	}
+	return
+}
+
+func hasHttpsForwarded(r *http.Request) bool {
+	if parseForwardedHeader(r, "proto")[0] == "https" {
+		return true
```

Comment:
could this lead to an index out of bounds exception if `parseFormwardedHeader` returns an empty array due to lack of `["Forwarded"]` header?